src/test/java/cloud/eppo/ProfileBaseEppoClientTest.java

```diff
@@ -0,0 +1,94 @@
+package cloud.eppo;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import cloud.eppo.logging.Assignment;
+import cloud.eppo.logging.AssignmentLogger;
+import cloud.eppo.ufc.dto.Attributes;
+import java.lang.management.ManagementFactory;
+import java.lang.management.ThreadMXBean;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ProfileBaseEppoClientTest {
+  private static final Logger log = LoggerFactory.getLogger(ProfileBaseEppoClientTest.class);
+
+  private static final String DUMMY_FLAG_API_KEY = "dummy-flags-api-key"; // Will load flags-v1
+  private static final String TEST_HOST =
+      "https://us-central1-eppo-qa.cloudfunctions.net/serveGitHubRacTestFile";
+
+  private static BaseEppoClient eppoClient;
+  private static final AssignmentLogger noOpAssignmentLogger =
+      new AssignmentLogger() {
+        @Override
+        public void logAssignment(Assignment assignment) {
+          /* no-op */
+        }
+      };
+
+  @BeforeAll
+  public static void initClient() {
+    eppoClient =
+        new BaseEppoClient(
+            DUMMY_FLAG_API_KEY,
+            "java",
+            "3.0.0",
+            TEST_HOST,
+            noOpAssignmentLogger,
+            null,
+            false,
+            false);
+
+    eppoClient.loadConfiguration();
+
+    log.info("Test client initialized");
+  }
+
+  @Test
+  public void testGetStringAssignment() {
+    Map<String, AtomicInteger> variationCounts = new HashMap<>();
+    Attributes subjectAttributes = new Attributes();
+    subjectAttributes.put("country", "FR");
+
+    ThreadMXBean threadBean = ManagementFactory.getThreadMXBean();
+    long startTime = threadBean.getCurrentThreadCpuTime();
+
+    int numIterations = 10000;
+
+    for (int i = 0; i < numIterations; i++) {
```

Comment:
These 10,000 iterations take ~70 ms to run on my machine, so this is still a fast (enough) test.